android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -1572,10 +1575,29 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
 
         if (KMManager.shouldAllowSetKeyboard()) {
           if (SystemKeyboard.keyboardPickerEnabled) {
+            KeyguardManager km = (KeyguardManager) appContext.getSystemService(Context.KEYGUARD_SERVICE);
+            if(km.inKeyguardRestrictedInputMode()) {
+              // Override system keyboard globe key action if screen is locked:
+              // 1. Switch to next Keyman keyboard (no menu)
+              // 2. When all the Keyman keyboards have been cycled through, advance to the next system keyboard
+              if (sysKbIndexOnLockScreen == -1) {
+                sysKbIndexOnLockScreen = getCurrentKeyboardIndex(context);
+                sysKbGlobeKeyAction = GlobeKeyAction.GLOBE_KEY_ACTION_SWITCH_TO_NEXT_KEYBOARD;
+              } else if (sysKbIndexOnLockScreen == getCurrentKeyboardIndex(context)) {
+                sysKbGlobeKeyAction = GlobeKeyAction.GLOBE_KEY_ACTION_ADVANCE_TO_NEXT_SYSTEM_KEYBOARD;
```

Comment:
To make sure I understand the logic here:

- When the system keyboard loads on the lock screen, we note its index in the installed keyboard list.
- We then cycle through them (modulo style) until we return to that initial index - at which point we then jump to the next input method instead.
  - It doesn't matter if that noted keyboard is technically the last keyboard on Keyman's list - it's now the 'starting' keyboard on the lock screen.

If so, I'm a bit worried that the shift in paradigm might be confusing - users would probably be used to Keyman swapping out early with the system keyboard when not on the lock screen.

Reply:
That said... I see Marc approved it as was in the 'stable' version, so... LGTM, I guess.

Reply:
Yes, the usage there may differ from the normal picker but because we are in a UI-constrained situation, the idea of pressing the globe button repeatedly until they get to the keyboard they need to type their password is probably sufficient.